### ISSUE TITLE
Target type in jsonPath method of MockMvcResultMatchers

### ISSUE THREAD

**Issue by juanmbellini** [spring-projects/spring-framework#23121]
**Affects:** 5.0.5

---

**This in an enhancement**

In #21129 `MockMvcResultMatchers` adds the possibility to set the `targetType` when using `jsonPath`, but this can only be achieved as in the following code:

```java
jsonPath("$.id").value(is(entity.getId()), Long.class) // getId returns a Long
```

It would be great to have the following method in the `MockMvcResultMatchers` class:

```java
<T> jsonPath(String, Matcher<T>, Class<T>) // here you can set the target type in the same method.
```

See https://github.com/spring-projects/spring-framework/blob/0a77477d325db2e5b0c83601d3c0bc336f7b506c/spring-test/src/main/java/org/springframework/test/web/servlet/result/MockMvcResultMatchers.java#L203

The new method can be something like this:

```java
public static <T> ResultMatcher jsonPath(String expression, Matcher<T> matcher, Class<T> targetType) {
    return new JsonPathResultMatchers(expression).value(matcher, targetType);
}
```

**Comment by rustyTheClone** [comment]
Solved in pull-request #23141 
I added a method `MockMvcResultMatchers#jsonPath(String, Class<T>, Matcher<T>)`

**Comment by sbrannen** [comment]
Superseded by PR #23141
